### PR TITLE
[TK-05277] Kitsune transport api updates / prep for proxy

### PR DESCRIPTION
**Pull request by neonphog**
Managing the Listeners separate from the Connections proved too complicated to wrangle especially for the proxy code.

This simplification is a complexity trade-off - It simplifies the usage of the API at the expense of less control over individual channels and stricter requirements on the implementations of low-level transports. In the case of the proxy code which is *not* a low-level transport, but rather composes other low-level transports, it makes a huge difference to code readability / implementation complexity.

Oddly enough, this also required less lines of code as evidenced by 518 deletions vs 350 additions...

Walkthrough Video (1:51): https://www.youtube.com/watch?v=cj2MrKoj2FQ

### CODE REVIEW

File: crates/kitsune_p2p/transport_quic/src/listener.rs [hunk 1]
```diff
@@ -1,26 +1,114 @@
...
-    transport::transport_connection::*,
-    transport::transport_listener::*,
+    dependencies::{ghost_actor, ghost_actor::GhostControlSender, url2::*},
     transport::*,
 };
-use std::net::SocketAddr;
+use std::{collections::HashMap, net::SocketAddr};
+
+/// Convert quinn async read/write streams into Vec<u8> senders / receivers.
+fn tx_bi_chan(
```
Comment (freesig) [inline crates/kitsune_p2p/transport_quic/src/listener.rs:9]: Is the point of this function to basically map the quinn streams to futures channels?

File: crates/kitsune_p2p/transport_quic/src/listener.rs [hunk 2]
```diff
@@ -35,6 +123,61 @@ impl ListenerInnerHandler for TransportListenerQuic {
             .map_err(TransportError::other)?;
         Ok(async move { Ok(out) }.boxed().into())
     }
+
+    fn handle_take_connecting(
```
Comment (freesig) [inline crates/kitsune_p2p/transport_quic/src/listener.rs:127]: Some comments inside this function might be helpful. It's pretty hard to follow what is happing

File: crates/kitsune_p2p/transport_quic/src/listener.rs [hunk 3]
```diff
@@ -51,18 +194,30 @@ impl TransportListenerHandler for TransportListenerQuic {
         Ok(async move { Ok(out) }.boxed().into())
     }
 
-    fn handle_connect(
+    fn handle_create_channel(
```
Comment (freesig) [inline crates/kitsune_p2p/transport_quic/src/listener.rs:197]: Maybe this one as well. I get the gist that it's taking in some quinn stream and creating the futures channels but I don't really understand the  flow. I think a few little comments would help especially if we are ever chasing bugs through this code in the future.

**Review by freesig — APPROVED** [review]
This looks good to me but I find it a little hard to actually follow what's going on in the code. Maybe some comments would help

File: crates/kitsune_p2p/transport_quic/src/listener.rs (hunk 2 above)
Comment (neonphog) [reply crates/kitsune_p2p/transport_quic/src/listener.rs:129]: :+1: done

File: crates/kitsune_p2p/transport_quic/src/listener.rs (hunk 3 above)
Comment (neonphog) [reply crates/kitsune_p2p/transport_quic/src/listener.rs:211]: :+1: done
